python/lsst/summit/utils/plotting.py

```diff
@@ -0,0 +1,256 @@
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import numpy as np
+
+from lsst.afw.detection import FootprintSet, Footprint
+# from lsst.afw.table import SourceCatalog
+import lsst.geom as geom
+from lsst.summit.utils import getQuantiles
+import lsst.afw.image as afwImage
+
+import logging
+
+import matplotlib.pyplot as plt
+import matplotlib.colors as colors
+from mpl_toolkits.axes_grid1.axes_divider import make_axes_locatable
+
+import astropy.visualization as vis
+
+
+def plot(inputData,
+         figure=None,
+         centroid=None,
+         footprints=None,
+         sourceCat=None,
+         title=None,
+         showCompass=False,
+         stretch='linear',
+         percentile=99.,
+         cmap='gray_r',
+         compassLocation=250,
+         addLegend=False,
+         savePlotAs=None,
+         logger=None):
+
+    """ Plot an input image accommodating different data types and additional
+    features, like: overplotting centroids, compass (if the input image
+    has WCS), stretching, plot title, and legend.
+
+    Parameters
+    ----------
+    inputData : `numpy.array` or
+                `lsst.afw.image.Exposure` or
+                `lsst.afw.image.Image`, or
+                `lsst.afw.image.MaskedImage`
+        The input data.
+        Defaults to 'image'.
+    figure : `matplotlib.figure.Figure`, optional
+         The Matplotlib figure that will be used for plotting.
+    centroid : `list`
+        The centroids parameter as a list of tuples.
+        Each tuple is a centroid with its (X,Y) coordinates.
+    footprints:  `lsst.afw.detection.FootprintSet` or
+                 `lsst.afw.detection.Footprint` or
+                 `list` of `lsst.afw.detection.Footprint`
+        The footprints containing centroids to plot.
+    sourceCat: `lsst.afw.table.SourceCatalog`:
+        A lsst.afw.table.SourceCatalog object containing centroids
+        to plot.
+    title : `str`, optional
+        Title for the plot.
+    showCompass : `bool`, optional
+        Add compass to the plot? Defaults to False.
+    stretch : `str', optional
+        Changes mapping of colors for the image. Avaliable options:
+        ccs, log, power, asinh, linear, sqrt. Defaults to linear.
+    percentile : `float', optional
+        Parameter for astropy.visualization.PercentileInterval.
+        Sets lower and upper limits for a stretch.
```

Comment:
Could you just note that this is ignore for the `ccs` option please?

Reply:
Sure.